File: tutorials/code/simp/cpp/eval-auto/eval-auto-print.cpp
#include <linkpred.hpp>
#include <iostream>
using namespace LinkPred::Simp;
int main() {
	int nbRuns = 10;
	double edgeRemRatio = 0.1; 
	// Create an evaluator object
	Evaluator eval;
	// Add predictors to be evaluated
	eval.addCNE();
	eval.addADA();
	eval.addKAB();
	// Add performance measures
	eval.addROC();
	eval.addTPR();
	// Run experiment on the specified network
	eval.run("Zakarays_Karate_Club.edges", nbRuns, edgeRemRatio);
	// Print the header row
	auto res = eval.getPerfRes(0);
	for (auto it =  res.begin(); it != res.end(); ++it) {
		std::cout << it->name << "\t" ;
	}
	std::cout << "\n";
	// Print the results of each iteration
	for(int i = 0; i < nbRuns; i++) {
		auto res = eval.getPerfRes(i);
		for (auto it =  res.begin(); it != res.end(); ++it) {
			std::cout << it->res << "\t";
		}
		std::cout << "\n";
	}
	return 0;
}

File: tutorials/code/simp/python/eval-auto/eval-auto-print.py
# Import the module
import LinkPredPython as lpp
nbRuns = 10;
edgeRemRatio = 0.1; 
# Create an evuator object
ev = lpp.Evaluator();
# Add predictors to be evuated
ev.addCNE();
ev.addADA();
ev.addKAB();
# Add performance measures
ev.addROC();
ev.addTPR();
# Run experiment on the specified network
ev.run("Zakarays_Karate_Club.edges", nbRuns, edgeRemRatio);
import sys # For printing
# Print the header row
res = ev.getPerfRes(0);
for r in res:
	sys.stdout.write(r.name + "\t");
sys.stdout.write("\n");
# Print the results of each iteration
for i in range(nbRuns):
	res = ev.getPerfRes(i);
	for r in res:
		sys.stdout.write("{:.4f}".format(r.res) + "\t");
	sys.stdout.write("\n");
